ext/random/random.c

```diff
@@ -92,7 +92,7 @@ static inline uint32_t rand_range32(const php_random_algo *algo, php_random_stat
 	total_size = 0;
 	do {
 		r = algo->generate(status);
-		result = (result << (8 * status->last_generated_size)) | r;
+		result = result | (r << (total_size * 8));
```

Comment:
Not sure if
```suggestion
		result |= r << (total_size * 8);
```
would be "better", "worse", or just "not worth it"? 😅

Reply:
The resulting object code should be identical. In this case I've opted to write this out explicitly to make the `|` a little more visible compared to the compound assignment.

Searching through the entire source tree with [Coccinelle](https://coccinelle.gitlabpages.inria.fr/website/) reveals that it's rare to spell it out explicitly, but there's precedent.

[1]
```
@@
identifier var;
expression e;
binary operator o;
@@

* var = (var) o (e);
```

Reply:
I wasn't concerned about the resulting object code, but rather to factorize duplication in the source code (when possible). But in this case I wasn't sure; the expanded form is more "visual" indeed 🙂 (but it seems I can't mark this thread as resolved?)

PS: Coccinelle looks cool! (*les Français font parfois des trucs bien aussi* 😄)